server/src/thread/model.rs

```diff
@@ -350,6 +351,95 @@ impl Threads {
 
         Ok(comment)
     }
+
+    pub async fn update_comment<'a, E>(
+        payload: &update_comment_mutation::Payload,
+        executor: E,
+    ) -> Result<operation::SuccessOutput, operation::Error>
+    where
+        E: Executor<'a>,
+    {
+        if payload.content.is_empty() {
+            return Err(operation::Error::BadRequest {
+                reason: "content is empty".to_string(),
+            });
+        }
+
+        let mut transaction = executor.begin().await?;
+
+        let is_trashed = match sqlx::query!(
+            r#"
+                SELECT trashed FROM uuid WHERE id = ?
+            "#,
+            payload.comment_id
+        )
+        .fetch_one(&mut transaction)
+        .await
+        {
+            Ok(record) => record.trashed,
+            Err(_) => {
```

Comment:
We could map exactly the error (it should be sqlx:Error:RowNotFound and throw it accordingly. The others error should be a server error.